### PR TITLE
Use Java servlets to personalize page

### PR DESCRIPTION
**Pull request by peter-parque**
Add an input box for users to get a personalized hello, generated by the Java servlet.

### CODE REVIEW

**Comment by peter-parque** [comment]
[Live here!](https://ptchoz-step-2020.appspot.com/) The box at the bottom is newest.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -25,8 +26,10 @@
 public class DataServlet extends HttpServlet {
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String text = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
+    String name = text.substring(text.indexOf(":")+2, text.lastIndexOf("\""));
```
Comment (EmeraldBaldwin) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Please comment on what this is parsing. What is the name expected to look like?

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -26,3 +26,22 @@ function addRandomGreeting() {
...
+  const response = await fetch('/data', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(data),
+  });
+  const text = await response.text();
+  console.log(text);
+  document.getElementById('personal-hello-containter').innerText = text;
```
Comment (EmeraldBaldwin) [inline portfolio/src/main/webapp/script.js]: typo?

**Review by EmeraldBaldwin — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (peter-parque) [reply portfolio/src/main/webapp/script.js]: fixed!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (peter-parque) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: I had this before using GSON to generate JSON with Java - fixed in db5f6e7d2254414a34fc04c652bdb21abb9d2e3a

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -14,7 +14,9 @@
 
 package com.google.sps.servlets;
 
-import java.util.stream.Collectors;
+import com.google.gson.*;
```
Comment (EmeraldBaldwin) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:23]: Please order imports alphabetically

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -25,11 +27,40 @@
...
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String jsonComments = new Gson().toJson(comments);
+
+        response.setContentType("text/html;");
+        response.getWriter().println(jsonComments);
+    }
+}
+
+class Comment {
```
Comment (EmeraldBaldwin) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:90]: I like this a lot!

I'm not suggesting you do this, but have you seen the Builder pattern? It's an interesting tool to have: https://dzone.com/articles/design-patterns-the-builder-pattern

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -30,18 +30,37 @@ function addRandomGreeting() {
...
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify(data),
-  });
-  const text = await response.text();
-  console.log(text);
-  document.getElementById('personal-hello-containter').innerText = text;
+// async function getPersonalHello() {
```
Comment (EmeraldBaldwin) [inline portfolio/src/main/webapp/script.js]: In general, try not to commit commented code.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (peter-parque) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:23]: done!

Comment (peter-parque) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:23]: Is this part of Google's style guides or generally accepted ones? I don't have access to go/codestyles to check

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (peter-parque) [reply portfolio/src/main/webapp/script.js]: Fixed!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (EmeraldBaldwin) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:23]: This is a part of Google's style guides: https://google.github.io/styleguide/

**Review by EmeraldBaldwin — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (peter-parque) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:90]: I hadn't seen that before! I implemented it to get a feel for it, and I really like it.

**Pull request merged**